pkg/worker/worker.go

```diff
@@ -0,0 +1,278 @@
+/*
+Package worker allows to start, stop and query Linux processes as concurrent jobs.
+
+Each job is saved to a provided storage via DataStorage interface. Its progress
+is updated each time a new line in StdIn or StdErr appears.
+
+To make use of the package create a new Worker using its constructor - NewWorker
+and provide a correct DataStorage. Control the processes using Start() and Stop()
+methods. Query the output anytime, even when the process is running using Query()
+method.
+
+DataStorage interface must include working data synchronization methods: Lock()
+and Unlock().
+*/
+package worker
+
+import (
+	"io"
+	"log"
+	"os/exec"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/jinzhu/copier"
+	"github.com/pkg/errors"
+
+	googleUUID "github.com/google/uuid"
+)
+
+var (
+	// ErrJobNotFound indicates that the requested job
+	// has not been found in the DataStorage.
+	ErrJobNotFound = errors.New("job has not been found")
+
+	// ErrJobAlreadyStopped indicates that the requested job
+	// has already been stopped.
+	ErrJobAlreadyStopped = errors.New("job has already been stopped")
+)
+
+// DataStorage allows to store and retrieve jobs from any storage that implements the interface.
+// The storage needs to implement sync.Mutex or other form of access synchronization based on
+// Lock() and Unlock() methods.
+type DataStorage interface {
+	Store(job Job) error
+	Retrieve(uuid googleUUID.UUID) (Job, error)
+}
+
+// MemoryStorage is a simple, in-memory implementation of the DataStorage interface.
+// Avoid using in production. Implement DataStorage with database of your choice.
+type MemoryStorage struct {
+	Data map[googleUUID.UUID]Job
+	sync.Mutex
```

Comment:
don't embed the mutex to avoid exposing it to the callers
use a private field instead